tutorials/single-page-tutorials/nested-nfts-with-wallet-lib.md

```diff
@@ -0,0 +1,423 @@
+const fs = require('fs');
+
+// Environment variables
+require('dotenv').config({ path: './.env' });
+const password = process.env.SH_PASSWORD;
+const accountName = process.env.ACCOUNT_NAME;
+
+// Network configuration
+const { networkConfig } = require('./networkConfig.js');
+const explorerURL = networkConfig.explorer;
+
+// For better readability, some console output will be printed in a different color
+const consoleColor = '\x1b[36m%s\x1b[0m';
+```
+
+#### 2. Start local IPFS node
+
+The function `startIpfsNode()` starts a local IPFS node for file upload.
+
+```javascript
+// This function starts a local IPFS node for file upload
+async function startIpfsNode() {
+  console.log('\n');
+  console.log(consoleColor, `Start local IPFS node for upload:`);
+
+  let node;
+  if (!node) {
+    node = await IPFS.create({
+      repo: `ipfs_node`,
+    });
+  }
+  return node;
+}
+```
+
+#### 3. Upload file to IPFS
+
+The function `uploadToIPFS()` receives a running ipfs node instance and uploads a file to IPFS from a given file path.
+
+```javascript
+// This function receives a running ipfs node instance and a file path for the upload
+async function uploadToIPFS(ipfsNode, filePath) {
+  try {
+    // Read file from path
+    const file = fs.readFileSync(filePath);
+
+    // Upload file to IPFS
+    const fileAdded = await ipfsNode.add(file);
+    const contentIdentifier = fileAdded.path;
+
+    console.log('\n');
+    console.log(consoleColor, `Your file was successfully uploaded to IPFS:`);
+    console.log(`https://ipfs.io/ipfs/${contentIdentifier}`, '\n');
+
+    return contentIdentifier;
+  } catch (error) {
+    console.error('IPFS upload error', error);
+  }
+}
+```
+
+#### 4. Mint NFT
+
+The function `mintNFT()` takes a NFT name, the previously created and unique IPFS identifier and an unlocked Stronghold account in order to mint an NFT on the configurated network. The metadata will be stored in the immutable metadata field, meaning it will remain unchanged for as long as the NFT itself exists on the network. Also note that the function does not pass a specific address to the Stronghold Account method `mintNfts()`. By default the method will mint all NFTs on the first address of the used Stronghold account, as long as it holds enough funds for Storage Deposit.
+
+:::note
+
+The NFT metadata follows [Tangle Improvement Proposal (TIP 27)](https://github.com/Kami-Labs/tips/blob/main/tips/TIP-0027/tip-0027.md).
+
+:::
+
+```javascript
+// This function creates NFT metadata and mints a new NFT
+async function mintNFT(itemName, ipfsCid, StrongholdAccount) {
+  // Define NFT metadata
+  const metadataObject = {
+    standard: 'IRC27',
+    type: 'image/jpeg',
+    version: 'v1.0',
+    name: `${itemName}`,
+    uri: `https://ipfs.io/ipfs/${ipfsCid}`,
+  };
+
+  const metadataBytes = utf8ToHex(JSON.stringify(metadataObject));
+
+  const response = await StrongholdAccount.mintNfts([
+    {
+      immutableMetadata: metadataBytes,
+    },
+  ]);
```

Comment:
We need to wait for the next milestone and sync to prevent the "no funds" issue.